lib/cereal/patterns/chain.go

```diff
@@ -0,0 +1,333 @@
+package patterns
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"github.com/chef/automate/lib/cereal"
+)
+
+var ErrIncorrectParameters = errors.New("Incorrect number of parameters given")
+var ErrTaskWorkflowInvalid = errors.New("Could not determine workflow for task")
```

Comment:
potential follow up: move these to error types where we can add more information to them